client/src/api.ts

```diff
@@ -158,6 +159,68 @@ export async function fetchSubjects(
   return result.data.data.subjects;
 }
 
+export async function fetchUsers(
+  searchParams?: SearchParams
+): Promise<Connection<User>> {
+  const params = { ...defaultSearchParams, ...searchParams };
+  const { filter, limit, cursor, sortBy, sortAscending } = params;
+  const result = await graphqlRequest.post("", {
+    query: `
+    query FetchUsers($filter: Object!, $limit: Int!, $cursor: String!, $sortBy: String!, $sortAscending: Boolean!){
+      fetchUsers(filter: $filter, limit: $limit,cursor: $cursor,sortBy: $sortBy,sortAscending: $sortAscending){
+        edges {
+          node {
+            _id
+            name
+            email
+            userRole
+            defaultMentor{
+              _id
+            }
+          }
+        }
+        pageInfo {
+          startCursor
+          endCursor
+          hasPreviousPage
+          hasNextPage
+        }
+      }
+    }`,
+    variables: {
+      filter: JSON.stringify(filter),
+      limit,
+      cursor,
+      sortBy,
+      sortAscending,
+    },
+  });
+  return result.data.data;
+}
+
+export async function updateUserPermissions(
+  userId: string,
+  permissionLevel: string,
+  accessToken: string
+): Promise<User> {
+  const headers = { Authorization: `bearer ${accessToken}` };
+  const result = await graphqlRequest.post(
+    "",
+    {
+      query: `mutation UpdateUserPermissions($userId: String!, $permissionLevel: String!){
+        me {
+          updateUserPermissions(userId: $userId, permissionLevel: $permissionLevel) {
+            userRole
+          }
+        }
+      }`,
+      variables: { userId, permissionLevel },
+    },
+    { headers: headers }
+  );
+  return result.data.data.me.updateUserPermissions;
```

Comment:
check for errors (have a test for how errors are handled and then make it work here)